test/autorest/generated/lrogroup/pollers.go

```diff
@@ -11,4536 +11,73 @@ import (
-	if err != nil {
-		return nil, err
-	}
-	return result, nil
-}
-
-// ResumeToken generates the string token that can be used with the ResumeLrosaDsPostAsyncRelativeRetryNoPayloadPoller method
-// on the client to create a new poller from the data held in the current poller type
-func (p *lrosaDsPostAsyncRelativeRetryNoPayloadPoller) ResumeToken() (string, error) {
-	if p.pt.hasTerminated() {
-		return "", errors.New("cannot create a ResumeToken from a poller in a terminal state")
-	}
-	js, err := json.Marshal(p.pt)
-	if err != nil {
-		return "", err
-	}
-	return string(js), nil
-}
-
-// Wait will continue to poll until a terminal state is reached or an error is encountered. Wait will use the
-// duration specified in the retry-after header, if the header is not specified then the pollingInterval that
-// is specified will be used to wait between polling requests.
-func (p *lrosaDsPostAsyncRelativeRetryNoPayloadPoller) Wait(ctx context.Context, pollingInterval time.Duration) (*LrosaDsPostAsyncRelativeRetryNoPayloadResponse, error) {
-	for p.Poll(ctx) {
-		if delay := p.response().RetryAfter(); delay > 0 {
-			time.Sleep(delay)
-		} else {
-			time.Sleep(pollingInterval)
-		}
-	}
-	return p.Response()
-}
-
-// response returns the last HTTP response.
-func (p *lrosaDsPostAsyncRelativeRetryNoPayloadPoller) response() *azcore.Response {
-	return p.pt.latestResponse()
-}
-
-// LrosaDsPostNonRetry400Poller provides polling facilities until the operation completes
-type LrosaDsPostNonRetry400Poller interface {
-	Poll(context.Context) bool
-	Response() (*LrosaDsPostNonRetry400Response, error)
-	ResumeToken() (string, error)
-	Wait(ctx context.Context, pollingInterval time.Duration) (*LrosaDsPostNonRetry400Response, error)
-}
-
-type lrosaDsPostNonRetry400Poller struct {
-	// the client for making the request
-	client *lrosaDsOperations
-	// polling tracker
-	pt pollingTracker
-}
-
-// Poll returns false if there was an error or polling has reached a terminal state
-func (p *lrosaDsPostNonRetry400Poller) Poll(ctx context.Context) bool {
-	return !lroPollDone(ctx, p.client.p, p.pt)
+	return nil, p.pt.pollingError()
 }
 
-// Response returns the latest response that is stored from the latest polling operation
-func (p *lrosaDsPostNonRetry400Poller) Response() (*LrosaDsPostNonRetry400Response, error) {
-	if p.pt.pollingError() != nil {
-		return nil, p.pt.pollingError()
-	}
-	resp := p.response()
-	if resp == nil {
-		return nil, errors.New("did not find a response on the poller")
+func (p *httpPoller) FinalResponse(ctx context.Context) (*HttpResponse, error) {
+	if p.pt.finalGetURL() == "" {
+		// we can end up in this situation if the async operation returns a 200
+		// with no polling URLs.  in that case return the response which should
+		// contain the JSON payload (only do this for successful terminal cases).
+		if lr := p.pt.latestResponse(); lr != nil && p.pt.hasSucceeded() {
+			result, err := httpPollerHandleResponse(lr, p.pipeline)
```

Comment:
Won't we be calling the handler on the client?

Reply:
Well we could but the handlers are specific to the poller response types. I think it will reduce code for it to be part of the pollers generation. Since the response handling should be the same for responses that use the same models. I could add it on the poller. Or why do you think it should be on the client?

Reply:
It's already on the client though right?  I was suggesting we use the one that's already there.

Reply:
Well the ones that are on the client are like xOperationHandleResponse (like delete202HandleResponse), but since these pollers need to be common among all operations, then we cant really use each operation's handle response method.

Reply:
Presumably because of the different set of status codes for success.  We can place a func on the poller that's set to the operation's response handler as they should all have the same signature, just different bodies.

Reply:
Ok, will do.

Reply:
Done